### PR TITLE
Evaluator improvements

### PR DESCRIPTION
**Pull request by mnonnenmacher**
Please see the commit messages for details.

---
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/1466)

### CODE REVIEW

File: evaluator/src/main/kotlin/PackageRule.kt [hunk 1]
```diff
@@ -135,6 +135,18 @@ open class PackageRule(
          */
         val licenseFindings: Map<LicenseFinding, List<PathExclude>> = emptyMap()
     ) : Rule(ruleSet, name) {
+        /**
```
Comment (sschuberth) [inline evaluator/src/main/kotlin/PackageRule.kt]: Commit message nit: "be accessed".

File: evaluator/src/main/kotlin/RuleSet.kt [hunk 2]
```diff
@@ -66,7 +66,11 @@ class RuleSet(val ortResult: OrtResult) {
             scope: Scope,
             project: Project
         ) {
-            val pkg = ortResult.analyzer?.result?.packages?.find { it.pkg.id == pkgRef.id }?.pkg!!
+            val pkg = ortResult.analyzer?.result?.packages?.find { it.pkg.id == pkgRef.id }?.pkg
+                ?: ortResult.analyzer?.result?.projects?.find { it.id == pkgRef.id }?.toPackage()
```
Comment (sschuberth) [inline evaluator/src/main/kotlin/RuleSet.kt]: Slightly less duplication:

            val pkg = ortResult.analyzer?.result?.let { result ->
                result.packages.find { it.pkg.id == pkgRef.id }?.pkg
                    ?: result.projects.find { it.id == pkgRef.id }?.toPackage()
            }

File: evaluator/src/main/kotlin/PackageRule.kt (hunk 1 above)
Comment (mnonnenmacher) [reply evaluator/src/main/kotlin/PackageRule.kt]: Done.

File: evaluator/src/main/kotlin/RuleSet.kt (hunk 2 above)
Comment (mnonnenmacher) [reply evaluator/src/main/kotlin/RuleSet.kt]: Done.

**Comment by mnonnenmacher** [comment]
@sschuberth : I have added one more commit to this PR with an important performance improvement.

File: evaluator/src/main/kotlin/PackageRule.kt [hunk 3]
```diff
@@ -104,11 +104,7 @@ open class PackageRule(
         val licenses = licenseView.licenses(pkg, detectedLicenses.map { it.license })
 
         licenses.forEach { (license, licenseSource) ->
-            val findings = if (licenseSource == LicenseSource.DETECTED) {
-                ruleSet.ortResult.collectLicenseFindings()[pkg.id].orEmpty()
-            } else {
-                emptyMap()
-            }
+            val findings = ruleSet.licenseFindings[pkg.id].orEmpty()
```
Comment (sschuberth) [inline evaluator/src/main/kotlin/PackageRule.kt]: Is dropping the condition intended here?

Comment (mnonnenmacher) [reply evaluator/src/main/kotlin/PackageRule.kt]: Thanks, fixed.

**Pull request merged**